wolfssl/wolfcrypt/kyber.h

```diff
@@ -0,0 +1,228 @@
+#endif /* WOLFSSL_KYBER512 */
+
+/* Kyber-768 parameters */
+#ifdef WOLFSSL_KYBER768
+/* Number of polynomials in a vector and vectors in a matrix. */
+#define KYBER768_K          3
+
+/* Size of a polynomial vector. */
+#define KYBER768_POLY_VEC_SZ             KYBER_POLY_VEC_SZ(KYBER768_K)
+/* Size of a compressed polynomial based on bits per coefficient. */
+#define KYBER768_POLY_COMPRESSED_SZ      KYBER_POLY_COMPRESSED_SZ(4)
+/* Size of a compressed vector polynomial based on dimensions and bits per
+ * coefficient. */
+#define KYBER768_POLY_VEC_COMPRESSED_SZ  \
+    KYBER_POLY_VEC_COMPRESSED_SZ(KYBER768_K, 10)
+
+/* Public key size. */
+#define KYBER768_PUBLIC_KEY_SIZE  \
+    (KYBER768_POLY_VEC_SZ + KYBER_SYM_SZ)
+/* Private key size. */
+#define KYBER768_PRIVATE_KEY_SIZE \
+    (KYBER768_POLY_VEC_SZ + KYBER768_PUBLIC_KEY_SIZE + 2 * KYBER_SYM_SZ)
+/* Cipher text size. */
+#define KYBER768_CIPHER_TEXT_SIZE \
+    (KYBER768_POLY_VEC_COMPRESSED_SZ + KYBER768_POLY_COMPRESSED_SZ)
+#endif /* WOLFSSL_KYBER768 */
+
+/* Kyber-1024 parameters */
+#ifdef WOLFSSL_KYBER1024
+/* Number of polynomials in a vector and vectors in a matrix. */
+#define KYBER1024_K         4
+
+/* Size of a polynomial vector. */
+#define KYBER1024_POLY_VEC_SZ             KYBER_POLY_VEC_SZ(KYBER1024_K)
+/* Size of a compressed polynomial based on bits per coefficient. */
+#define KYBER1024_POLY_COMPRESSED_SZ      KYBER_POLY_COMPRESSED_SZ(5)
+/* Size of a compressed vector polynomial based on dimensions and bits per
+ * coefficient. */
+#define KYBER1024_POLY_VEC_COMPRESSED_SZ \
+    KYBER_POLY_VEC_COMPRESSED_SZ(KYBER1024_K, 11)
+
+/* Public key size. */
+#define KYBER1024_PUBLIC_KEY_SIZE  \
+    (KYBER1024_POLY_VEC_SZ + KYBER_SYM_SZ)
+/* Private key size. */
+#define KYBER1024_PRIVATE_KEY_SIZE \
+    (KYBER1024_POLY_VEC_SZ + KYBER1024_PUBLIC_KEY_SIZE + 2 * KYBER_SYM_SZ)
+/* Cipher text size. */
+#define KYBER1024_CIPHER_TEXT_SIZE \
+    (KYBER1024_POLY_VEC_COMPRESSED_SZ + KYBER1024_POLY_COMPRESSED_SZ)
+#endif /* WOLFSSL_KYBER1024 */
+
+
+/* Maximum dimensions and sizes of supported key types. */
+#ifdef WOLFSSL_KYBER1024
+#define KYBER_MAX_K                 KYBER1024_K
+#define KYBER_MAX_PRIVATE_KEY_SIZE  KYBER1024_PRIVATE_KEY_SIZE
+#define KYBER_MAX_PUBLIC_KEY_SIZE   KYBER1024_PUBLIC_KEY_SIZE
+#define KYBER_MAX_CIPHER_TEXT_SIZE  KYBER1024_CIPHER_TEXT_SIZE
+#elif defined(WOLFSSL_KYBER768)
+#define KYBER_MAX_K                 KYBER768_K
+#define KYBER_MAX_PRIVATE_KEY_SIZE  KYBER768_PRIVATE_KEY_SIZE
+#define KYBER_MAX_PUBLIC_KEY_SIZE   KYBER768_PUBLIC_KEY_SIZE
+#define KYBER_MAX_CIPHER_TEXT_SIZE  KYBER768_CIPHER_TEXT_SIZE
+#else
+#define KYBER_MAX_K                 KYBER512_K
+#define KYBER_MAX_PRIVATE_KEY_SIZE  KYBER512_PRIVATE_KEY_SIZE
+#define KYBER_MAX_PUBLIC_KEY_SIZE   KYBER512_PUBLIC_KEY_SIZE
+#define KYBER_MAX_CIPHER_TEXT_SIZE  KYBER512_CIPHER_TEXT_SIZE
+#endif
+
+
+
+enum {
+    /* Types of Kyber keys. */
+    KYBER512  = 0,
+    KYBER768  = 1,
+    KYBER1024 = 2,
```

Comment:
do we need 90S here?

Reply:
At config time you choose one set or the other.